### PR TITLE
feat: Allow testing of single spec modules

### PR DESCRIPTION
**Pull request by felixrieseberg**
This PR allows you to run `npm run test -match=menu`, which will then only run spec modules matching `menu`. This is crazy useful for anyone who's working on tests that would otherwise be at the very end of the testing cycle.

Also: I've used the opportunity to upgrade the JavaScript in `index.html` to ES6 and StandardJS, in line with our coding standards.

### CODE REVIEW

**Comment by codebytere** [comment]
if i could hug this PR i would

File: spec/static/index.html [hunk 1]
```diff
@@ -11,74 +11,83 @@
...
       return {
-        log: function() {
-          args = Array.prototype.slice.call(arguments);
-          ipcRenderer.send('console.log', args);
-        },
-        error: function() {
-          args = Array.prototype.slice.call(arguments);
-          ipcRenderer.send('console.error', args);
+        log: () => {
+          const args = Array.prototype.slice.call(arguments)
```
Comment (alexeykuzmin) [inline spec/static/index.html]: Arrow function resolves `arguments` from the lexical scope, i.e. from here
```
global.__defineGetter__('console', function () {
                                            ^
```

File: spec/static/index.html [hunk 2]
```diff
@@ -11,74 +11,83 @@
...
+          ipcRenderer.send('console.log', args)
         },
+        error: () => {
+          const args = Array.prototype.slice.call(arguments)
```
Comment (alexeykuzmin) [inline spec/static/index.html]: ES2015 way is to use rest parameters:

```js
error: function(...args) {
  ipcRenderer.send('console.error', args)
}
```

**Comment by felixrieseberg** [comment]
We should probably also write this _hot trick_ down somewhere, but I'm not sure where. Any tips? New file?

**Comment by ckerr** [comment]
build-system-overview.md#tests sounds like a good place for this One Simple Trick

File: spec/static/index.html (hunk 2 above)
Comment (ckerr) [reply spec/static/index.html]: That is a thing of beauty. Nice!

**Comment by ckerr** [comment]
@felixrieseberg What do you think about `build-system-overview.md#tests`?

**Comment by felixrieseberg** [comment]
@ckerr @codebytere Happy new year! It took me a while, but I finally came around to adding some docs.

**Pull request merged**